### PR TITLE
Upgrade to repo-config v0.5.2

### PR DESCRIPTION
**Pull request by llucax**
All the changes are related to regenerating the files using the cookiecutter templates in repo-config v0.5.2.

- Remove extensions from replay file
- Add empty `Introduction` variable
- Bump repo-config version to v0.5.2.
- Add common pylint options
- Add common source paths for isort
- Add editorconfig file
- Use repo-config for linting examples in docstrings
- ci: Checkout submodules
- Add a workflow to make sure release notes are updated
- Don't ship development files in the source distribution
- Add and sort mkdocs markdown extensions
- mkdocs: Add comment with more info about cross-linking
- Bump setuptools version

### CODE REVIEW

File: mkdocs.yml [hunk 1]
```diff
@@ -79,6 +76,8 @@ markdown_extensions:
         - name: mermaid
           class: mermaid
           format: "!!python/name:pymdownx.superfences.fence_code_format"
+  - pymdownx.tabbed
+  - pymdownx.tasklist
```
Comment (daniel-zullo-frequenz) [inline mkdocs.yml:80]: it seems `pymdownx.superfences` was removed in the commit `Add and sort mkdocs markdown extensions`. Was this deliberately done?

**Review by daniel-zullo-frequenz — COMMENTED** [review]
I have only a question to check for and the PR needs rebasing to the latest v0.x.x

File: MANIFEST.in [hunk 2]
```diff
@@ -1,9 +1,13 @@
...
+exclude .gitignore
 exclude CODEOWNERS
+exclude CONTRIBUTING.md
+exclude mkdocs.yml
+exclude noxfile.py
 exclude src/conftest.py
 recursive-exclude .github *
+recursive-exclude docs *
 recursive-exclude tests *
-recursive-exclude benchmarks *
```
Comment (Marenz) [inline MANIFEST.in:7]: We no longer exclude the benchmarks?

**Review by Marenz — DISMISSED** [review]

File: mkdocs.yml (hunk 1 above)
Comment (llucax) [reply mkdocs.yml:80]: Yes, it was duplicated. I clarified it in the commit.

File: MANIFEST.in (hunk 2 above)
Comment (llucax) [reply MANIFEST.in:7]: ~Yes, from the source distribution package.~

I misunderstood, you are right, this shouldn't be removed. Good catch!

**Comment by llucax** [comment]
Rebased and fixed the wrong removal of the benchmarks exclusion.

**Comment by llucax** [comment]
It would be good to merge this soon (before merging other dependabot stuff) as I got a few conflicts when rebasing.

**Review by Marenz — APPROVED** [review]

**Pull request merged**